File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        d = {}
        set_ans = False

        for x in nums:
            d[x] = d.get(x, 0) + 1

        for x in d:
            if d[x] > 1:
                set_ans = True
                break
            else:
                set_ans = False
        
        if set_ans:
            return True
        else:
            return False

File: Data Structures & Algorithms/majority-element/submission-0.js
class Solution {
    majorityElement(nums) {
        let n = nums.length;
        let hash = {};
        for (let i=0; i<n; i++) {
            if (hash[nums[i]]) {
                hash[nums[i]] += 1;
            } else {
                hash[nums[i]] = 1;
            }
        }

        for (let i=0; i<n; i++) {
            if (hash[nums[i]] >= (n/2)) {
                let key = Object.keys(hash).find(key => hash[key] === hash[nums[i]])
                return key;
            }
        }
    }
}
